docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
 Architecture Overview
 =====================
 
+Runhouse is a Python library that allows any application to flexibly and powerfully utilize remote compute
+infrastructure by deploying and calling remote services on the fly. It is principally designed for Machine
+Learning-style workloads (online, offline, training, and inference), where the need for heterogeneous
+remote compute is frequent and flexibility is paramount to minimize costs.
+
+The basic hypothesis is that incorporating dynamic compute into the runtime of an application, like workflow
```

Comment:
Could simplify? 
_Incorporating dynamic compute into workflow orchestrators  (Airflow, Prefect) or distributed libraries (Ray, Spark) is far more disruptive and less flexible at every level (dev workflow, debugging, DevOps, infra) than calling a remote service from those applications. _

Reply:
+1 here, I didn't have a solution but found this sentence too heavy and hard to parse